src/models/helpers/goalCollaborator.js

```diff
@@ -211,16 +211,18 @@ const removeCollaboratorsForType = async (
   typeName,
   linkBack = undefined,
 ) => {
-  const filteredLinkBack = linkBack?.filter((lb) => lb);
-  if (!filteredLinkBack && filteredLinkBack.length > 0) return;
+  if (!linkBack) return;
+  const filteredLinkBack = linkBack.filter((lb) => lb);
+  if (filteredLinkBack.length === 0) return;
+
   // Extract the key-value pair from the linkBack object
-  const [[linkBackKey, linkBackValues]] = Object.entries(linkBack);
+  const [[linkBackKey, linkBackValues]] = Object.entries(filteredLinkBack);
```

Comment:
I think this only grabs the first value of the array returned by Object.entries. Is that what you're going for?

<img width="705" alt="Screenshot 2023-11-26 at 12 18 56 PM" src="https://github.com/HHS/Head-Start-TTADP/assets/17074357/7f319601-76d7-4b38-9d1c-d695d75c7fd1">